packages/provider/asset/index.js

```diff
@@ -102,14 +105,14 @@ export default class Asset extends Component {
   }
 
   /**
-   * Process prop changes.
+   * Properties have been updated.
    *
-   * @param {Object} nextProps New props that we received.
+   * @param {Object} prev Previous properties.
    * @private
    */
-  componentWillReceiveProps(nextProps) {
-    this.debug('receiving new props', nextProps);
-    this.fetch(nextProps);
+  componentDidUpdate(prev) {
+    const props = this.props;
+    if (props !== prev) this.fetch(props);
```

Comment:
Since this is object comparison, won't this always be unequal?

Reply:
The object would only change if `props` are actually changing, not when state is updated for example.